pkg/api/object.go

```diff
@@ -25,12 +25,41 @@ import (
 	"github.com/rook/rook/pkg/model"
 )
 
+const defaultObjectStoreName = "default"
+
+func (h *Handler) objectContext(r *http.Request) *rgw.Context {
+	storeName := defaultObjectStoreName
+	if name, ok := mux.Vars(r)["store"]; ok {
+		storeName = name
+	}
+
+	return rgw.NewContext(h.context, storeName, h.config.ClusterHandler.GetClusterInfo)
+}
+
 // CreateObjectStore creates a new object store in this cluster.
 // POST
 // /objectstore
 func (h *Handler) CreateObjectStore(w http.ResponseWriter, r *http.Request) {
+	var objectStore model.ObjectStore
+	err := json.NewDecoder(r.Body).Decode(&objectStore)
+	if err != nil {
+		logger.Errorf("Error parsing object store settings: %+v", err)
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	// only the default store is supported through the rest api
+	if objectStore.Name == "" {
+		objectStore.Name = defaultObjectStoreName
+	}
+	if objectStore.Port == 0 {
+		objectStore.Port = model.RGWPort
+	}
+	if objectStore.RGWReplicas == 0 {
+		objectStore.RGWReplicas = 1
```

Comment:
is 1 the default replica size if not specified? If so, it should be defined as a constant.

Reply:
sure